sveltekit-authentication/index.md

```diff
@@ -0,0 +1,449 @@
+---
+title: 'Authentication in SvelteKit with Directus'
+description: 'How to authenticate users and integrate state managment with Cookies using SvelteKit SSR'
+author:
+  name: 'Eike Thies'
+  avatar_file_name: 'avatar.jpg'
+---
+
+
+## Overview
+
+In this Guide we will show how to set up a complete authentication and authorization mechanism in SvelteKit using Directus as the Provider. The user session will be persisted via Server-Side Cookies and evidently be used within the Directus SDK to make authenticated Calls. As an example we will create a new Role showing how authorization works.
+
+
+## Primer Authentication
+
+There are different ways how to store the user session. Here we will use the most secure form via Server-Side Cookies using the `Same-Site=Strict` attribute which will prevent CSRF Attacks. This means in order to be able to access the Directus Backend from the client directly, the Directus Backend will need to live on the same domain as the SvelteKit Frontend. For local development this will not matter. For more Information on this topic see https://andrewlock.net/understanding-samesite-cookies/
```

Comment:
```suggestion
## Understanding Server-Side Cookies

A user session is...

There are different ways to store a user session within web applications. We will use the most secure - server-side cookies using the `Same-Site=Strict` attribute which will prevent CSRF attacks. CSRF attacks are..

To access Directus from the client directly with this approach, Directus will need to live on the same domain as the SvelteKit application. For local development, this will not matter as... 

[Learn more about SameSite cookies](https://andrewlock.net/understanding-samesite-cookies/).
```

This is excellent context but needs a little but more - each paragraph ends with a prompt for you to complete.

Reply:
i extended the user session and CSRF paragraphs. not sure if we should go too deeply into this in the article. i would rather link some external explanations let me know if in the new version it is still not enough.

i removed the "same domain" paragraph as it was wrong. actually it will work in any case, as the cookie is only for the frontend and the token will get read from there and send to the directus backend if needed via the Directus SDK. that makes it work for any case (directus living in the cloud or deployed yourself on your domain)